flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1082,6 +1110,13 @@ class JobManager(
 
         log.info(s"Using restart strategy $restartStrategy for $jobId.")
 
+        val jobMetrics = jobManagerMetricGroup match {
+          case Some(group) =>
+            group.addJob(jobGraph.getJobID, jobGraph.getName)
```

Comment:
That works as well and is probably safer.

I actually just run into this problem. Have you actually tried your changes out with submitting a job?